frontend/hub/access/resource-access/HubResourceAccessRoles.tsx

```diff
@@ -0,0 +1,159 @@
+import { RoleExpandedRow } from '../roles/components/RoleExpandedRow';
+
+function roleKeyFn(role: HubAccessRole) {
+  return role.name;
+}
+
+export function HubResourceAccessRoles(
+  props: PropsForHubRoleListForUser | PropsForHubRoleListForTeam
+) {
+  const { t } = useTranslation();
+  const getPageUrl = useGetPageUrl();
+  const { user, team, canEditAccess } = props;
+
+  const mappedRoles = useMemo(() => {
+    if (user) {
+      return user.object_roles?.map((role) => ({
+        name: role,
+      }));
+    } else if (team) {
+      return team.object_roles.map((role) => ({
+        name: role,
+      }));
+    }
+  }, [team, user]);
+
+  const tableColumns = useMemo<ITableColumn<HubAccessRole>[]>(
+    () => [
+      {
+        header: t('Role'),
+        cell: (role) => <TextCell text={role.name} />,
+        key: 'name',
+        value: (role) => role.name,
+        sort: 'name',
+        card: 'name',
+        list: 'name',
+      },
+    ],
+    [t]
+  );
+
+  const toolbarActions = useMemo<IPageAction<HubAccessRole>[]>(
+    () => [
+      {
+        type: PageActionType.Link,
+        selection: PageActionSelection.None,
+        isPinned: true,
+        icon: PlusCircleIcon,
+        label: t('Add role'),
+        isDisabled: canEditAccess
+          ? undefined
+          : t(
+              'You do not have permission to add a role. Please contact your system administrator if there is an issue with your access.'
+            ),
+        href: user
+          ? `${getPageUrl(HubRoute.NamespaceUserAccessAddRoles)}`
+          : `${getPageUrl(HubRoute.NamespaceTeamAccessAddRoles)}`,
+      },
+      { type: PageActionType.Seperator },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Multiple,
+        icon: TrashIcon,
+        label: t('Delete selected roles'),
+        onClick: () => alert(t('Todo')),
+        isDanger: true,
+      },
+    ],
+    [canEditAccess, getPageUrl, t, user]
+  );
+
+  const rowActions = useMemo<IPageAction<HubAccessRole>[]>(
+    () => [
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: TrashIcon,
+        label: t('Delete role'),
+        onClick: () => alert(t('Todo')),
+        isDisabled: canEditAccess
+          ? undefined
+          : t(
+              'You do not have permission to delete a role. Please contact your system administrator if there is an issue with your access.'
+            ),
+        isDanger: true,
+      },
+    ],
+    [canEditAccess, t]
+  );
+
+  const toolbarFilters = useMemo(() => {
+    const filters: IToolbarFilter[] = [
+      {
+        type: ToolbarFilterType.Text,
+        label: t('Role'),
+        key: 'name',
+        query: 'name',
+        comparison: 'contains',
+        placeholder: t('Filter by name'),
+        isPinned: true,
+      },
+    ];
+    return filters;
+  }, [t]);
+
+  const view = useInMemoryView<HubAccessRole>({
```

Comment:
@vidyanambiar I am  just starting to look at this PR. It does look pretty good.

I could not follow why we need to use the useInMemoryView, neither why the changes were required there. Is it something that we will need to use in order to use the new approach for breadcrumbs? 

Could we document this new hook, and how to use them?

Reply:
@nixocio Thanks for taking a look!

Reason for using the `useInMemoryView` hook: Unlike most other list UIs, this one does not have a direct API call for loading the users/teams table so we cannot utilize `usePulpView` or `useHubView` which are associated with an API call for fetching the data for the table. We need to user the `users` and `groups` array that is received within the namespace data response. So the table is constructed using these static arrays. The `useInMemoryView` is a hook provided in the framework to create a PageTable from static data. And it provides the utilities required like selectable rows, filtering etc. I needed to make a small update for making the filtering work correctly for text-based filters.